pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -125,6 +129,13 @@ func New(prov provider) (*Service, error) {
 	return svc, nil
 }
 
+// RegisterInvitationReceivedCallback registers callback function
+func (s *Service) RegisterInvitationReceivedCallback(fn func(*Invitation)) {
```

Comment:
Also - I'm curious why a callback function rather than a channel?

Reply:
`Will we need to consider unregister?` - yeah, it would be nice. I'll add it.
Will we need to eventually handle multiple registrations? - no, it seems to be redundant. 

I can replace it with a channel.

Reply:
Actually, it looks like I can use `RegisterMsgEvent`.

Reply:
Yes - Using the existing event mechanism is even better!